api/auth.py

```diff
@@ -0,0 +1,58 @@
+# This file is a part of GreedyBear https://github.com/honeynet/GreedyBear
+# See the file 'LICENSE' for copying permission.
+
+import logging
+from django.contrib.auth import login, logout
+from drf_spectacular.extensions import OpenApiAuthenticationExtension
+from durin import views as durin_views
+from durin.models import Client
+
+logger = logging.getLogger(__name__)
+
+
+""" Auth API endpoints """
+
+
+class LoginView(durin_views.LoginView):
+    def get_client_obj(self, request) -> Client:
+        client, _created = Client.objects.get_or_create(name="web-browser")
+        return client
+
+    def post(self, request, *args, **kwargs):
+        response = super(LoginView, self).post(request, *args, **kwargs)
+        uname = request.user.username
+        logger.info(f"LoginView: received request from '{uname}'.")
+        if request.user.is_superuser:
+            try:
+                # pass admin user's session
+                login(request, request.user)
+                logger.info(f"administrator:'{uname}' was logged in.")
+            except Exception:
+                logger.exception(f"administrator:'{uname}' login failed.")
+        return response
+
+
+class LogoutView(durin_views.LogoutView):
+    def post(self, request, *args, **kwargs):
+        uname = request.user.username
+        logger.info(f"perform_logout received request from '{uname}''.")
+        if request.user.is_superuser:
+            try:
+                logout(request)
+                logger.info(f"administrator: '{uname}' was logged out.")
+            except Exception:
+                logger.exception(f"administrator: '{uname}' session logout failed.")
+        return super(LogoutView, self).post(request, format=None)
+
+
+class DurinAuthenticationScheme(OpenApiAuthenticationExtension):
+    target_class = "durin.auth.CachedTokenAuthentication"
+    name = "durinAuth"
+
+    def get_security_definition(self, auto_schema):
+        return {
+            "type": "apiKey",
+            "in": "header",
+            "name": "Authorization",
+            "description": "Token-based authentication with required prefix: Token",
+        }
```

Comment:
Yaa I accept its a complete copy paste. I have gone through the complete documentation of durin . After getting an understanding of it. I saw the implementation of it in intel owl (as it mentioned to take it as reference) and it was completely the same code as it in examples of documentation expect somewhat good print statements and the schema at the bottom. Then I thought what would be a great thing than copy paste if its completely working fine in our case.

I am not clear of what all views I have to include in for the authentication(as there were 4 token based and 2 session based views in durin) but again Intelowl is the reference for me. so I thought why do I even add other views when they are enough.

Reply:
As mentioned, there is a disclaimer and there are licenses. You cannot just copy-paste only because you "thought that would be a great thing than copy paste if its completely working fine in our case." That is not how open source works